Add contenttype and contenttypemod Registries

SteveLLamb opened the pull request:
Closes #107 

A few notes on these registries:
- For contenttypemod the values for code and value in square brackets are meant to show it can variable and will be described as such on the web page. 
- For contenttypemod the metaType is enum values
- Regs are canonicalized, but set to manual order for initial order on page rendering (to replicate current page order)

palemieux commented on test/contenttypes.js:
```diff
@@ -0,0 +1,38 @@
+
+  it("invalid", () => {
+    assert.throw(() => validate([
+      "foo"
```
Maybe a better effort? :)

SteveLLamb replied:
Done, added in missing scope.

palemieux commented on test/contenttypemods.js:
```diff
@@ -0,0 +1,41 @@
+
+  it("invalid", () => {
+    assert.throw(() => validate([
+      "foo"
```
Recommend a more realistic bad example

SteveLLamb replied:
Done, added in missing metadata.

palemieux commented on src/main/data/contenttypes.json:
```diff
@@ -0,0 +1,146 @@
+  },
+  {
+    "code": "TLR",
+    "contentKind": {
```
_contentKind_ should be _cplContentKind_

SteveLLamb replied:
Updated.

palemieux commented on src/main/data/contenttypes.json:
```diff
@@ -0,0 +1,146 @@
+    "description": "A theatrical feature."
+  },
+  {
+    "code": "TLR",
```
_code_ should _dcncCode_

SteveLLamb replied:
Updated.

palemieux reviewed (changes requested):
What about `dcnc-content-mods` instead of `contenttypemod` to emphasize it is intended for dcnc only?

SteveLLamb commented:
> What about `dcnc-content-mods` instead of `contenttypemod` to emphasize it is intended for dcnc only?

Updated to `contentmodifiers` as data also includes metadata.

palemieux reviewed (changes requested):
Add `dcncSortOrder` integer to preserve historical ordering, and enforce it during validation.

SteveLLamb commented:
> Add `dcncSortOrder` integer to preserve historical ordering, and enforce it during validation.

Done, added 2 addition checks on Test (missing sort order, and sort order not an integer) and added sorting and duplication conditions on Validate.
